src/marshmallow/validate.py

```diff
@@ -70,7 +70,7 @@ def _regex_generator(self, relative: bool, require_tld: bool):
                         r")?"
                         if relative
                         else r"",  # host is optional, allow for relative URLs
-                        r"(?:/?|[/?]\S+)$",
+                        r"(?:/?|[/?]\S+)\Z",
```

Comment:
This line is also on 2.x branch, so I guess we ought to fix it there as well.

https://github.com/marshmallow-code/marshmallow/blob/2.x-line/src/marshmallow/validate.py

AFAICT, MA2's email validation is not impacted.

Reply:
I was able to reproduce the same issue on 2.x.

```py
>>> from marshmallow import Schema, fields
>>> 
>>> class Test(Schema):
...     foo = fields.Email()
... 
>>> Test().load({'foo': 'foo@foo.foo\n'})
UnmarshalResult(data={'foo': u'foo@foo.foo\n'}, errors={})
```

Reply:
It looks like `$` is still used for emails on the 2.x branch? https://github.com/marshmallow-code/marshmallow/blob/2.x-line/src/marshmallow/validate.py#L126

Reply:
I can create another PR for 2.x if ya'll want that?

Reply:
Good catch. I missed it.

> I can create another PR for 2.x if ya'll want that?

If you don't mind the extra work, that would be nice. The usual process is fix 2.x (if relevant) then merge into dev. Meanwhile, please keep this PR/branch as a reference for the merge into dev.

Thanks.